File: lachancuocgoi_kotline/app/src/main/java/com/example/lachancuocgoi/data/AlertHistoryEntry.kt
package com.example.lachancuocgoi.data

import androidx.compose.ui.graphics.Color
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

/**
 * Đại diện cho một lần hiển thị cảnh báo (batch hoặc immediate).
 * Được lưu trữ trong lịch sử cuộc gọi để tracking.
 */
data class AlertHistoryEntry(
    val timestamp: Long,                // Thời điểm hiển thị cảnh báo
    val analysisLevel: String,          // "L1", "L2", "L3"
    val riskLevel: String,              // "RED", "ORANGE"
    val alertCount: Int,                // Số cảnh báo trong batch (1 nếu L3)
    val displayedReason: String,        // Lý do cảnh báo được hiển thị
    val allReasons: List<String>? = null // Tất cả lý do trong batch (null nếu L3)
) {
    /**
     * Format timestamp thành HH:mm:ss
     */
    fun getFormattedTime(): String {
        val sdf = SimpleDateFormat("HH:mm:ss", Locale.getDefault())
        return sdf.format(Date(timestamp))
    }
    
    /**
     * Lấy màu tương ứng với risk level
     */
    fun getRiskLevelColor(): Color {
        return when (riskLevel) {
            "RED" -> Color(0xFFD32F2F)
            "ORANGE" -> Color(0xFFFF9800)
            else -> Color(0xFF9E9E9E)
        }
    }
    
    /**
     * Lấy icon emoji tương ứng
     */
    fun getRiskLevelIcon(): String {
        return when (riskLevel) {
            "RED" -> "🔴"
            "ORANGE" -> "🟠"
            else -> "⚪"
        }
    }
}

File: lachancuoicgoi_flutter_demo/lib/data/alert_history_entry.dart
import '../core/risk_level.dart';

class AlertHistoryEntry {
  const AlertHistoryEntry({
    required this.timestamp,
    required this.analysisLevel,
    required this.riskLevel,
    required this.alertCount,
    required this.displayedReason,
    this.allReasons,
  });

  final int timestamp;
  final String analysisLevel;
  final String riskLevel;
  final int alertCount;
  final String displayedReason;
  final List<String>? allReasons;

  factory AlertHistoryEntry.fromJson(Map<String, Object?> json) {
    return AlertHistoryEntry(
      timestamp: (json['timestamp'] as num?)?.toInt() ?? 0,
      analysisLevel: json['analysisLevel'] as String? ?? '',
      riskLevel: json['riskLevel'] as String? ?? 'GREEN',
      alertCount: (json['alertCount'] as num?)?.toInt() ?? 1,
      displayedReason: json['displayedReason'] as String? ?? '',
      allReasons: (json['allReasons'] as List?)?.whereType<String>().toList(),
    );
  }

  Map<String, Object?> toJson() {
    return {
      'timestamp': timestamp,
      'analysisLevel': analysisLevel,
      'riskLevel': riskLevel,
      'alertCount': alertCount,
      'displayedReason': displayedReason,
      'allReasons': allReasons,
    };
  }

  String getFormattedTime() {
    final time = DateTime.fromMillisecondsSinceEpoch(timestamp);
    return '${_pad(time.hour)}:${_pad(time.minute)}:${_pad(time.second)}';
  }

  /// ARGB color int for this entry's risk level, sourced from
  /// [RiskLevel.colorValue] (the canonical palette). The UI layer converts
  /// this to a Flutter `Color` — this data model stays Flutter-free.
  int getRiskLevelColor() => RiskLevel.fromString(riskLevel).colorValue;

  /// Stable icon key derived from the parsed [RiskLevel] level so the
  /// mapping logic lives in a single place.
  String getRiskLevelIcon() {
    final level = RiskLevel.fromString(riskLevel);
    return switch (level) {
      RiskLevel.red => 'red',
      RiskLevel.orange => 'orange',
      _ => 'neutral',
    };
  }

  static String _pad(int value) => value.toString().padLeft(2, '0');
}
